xbmc/windowing/wayland/WinSystemWaylandEGLContextGL.cpp

```diff
@@ -22,13 +22,35 @@
 
 #include <EGL/egl.h>
 
+#include "cores/VideoPlayer/VideoRenderers/LinuxRendererGL.h"
 #include "utils/log.h"
 
+#if defined(HAVE_LIBVA)
+#include "cores/VideoPlayer/DVDCodecs/Video/VAAPI.h"
+#include "cores/VideoPlayer/VideoRenderers/HwDecRender/RendererVAAPIGL.h"
+#endif
+
 using namespace KODI::WINDOWING::WAYLAND;
 
 bool CWinSystemWaylandEGLContextGL::InitWindowSystem()
 {
-  return CWinSystemWaylandEGLContext::InitWindowSystemEGL(EGL_OPENGL_BIT, EGL_OPENGL_API);
+  if (!CWinSystemWaylandEGLContext::InitWindowSystemEGL(EGL_OPENGL_BIT, EGL_OPENGL_API))
+  {
+    return false;
+  }
+
+  CLinuxRendererGL::Register();  
+  
+#if defined(HAVE_LIBVA)
+  bool general, hevc;
+  CRendererVAAPI::Register(GetVaDisplay(), m_eglContext.m_eglDisplay, general, hevc);
```

Comment:
I'm honestly not sure whether any Wayland compositor has support for multi-GPU currently. I also vaguely recall that X support for this is not that great and rather hacky?

Also, there's no protocol that could tell you what outputs correspond to which GPU - so even if a compositor one day supports a desktop that spans multiple GPUs, it would have to take care of uploading the rendered buffer you give to it to the right GPU by itself. That a window is being displayed on GPU 2 does not mean it has to be rendered by GPU 2. Depending on the hardware, the GPUs might even be able to share the buffers without copying (like Optimus).

In summary: This does not apply to Wayland, at least for the foreseeable future. There is also no code currently that would destroy/recreate the egl context. It's created in `InitWindowSystem` and destroyed in `DestroyWindowSystem`. I'd have to move that to a separate function first, I guess.

See also: Last paragraph [here](https://nouveau.freedesktop.org/wiki/MultiMonitorDesktop/)